tests/Makefile

```diff
@@ -48,44 +39,35 @@ keyboard_interface: cli/keyboard_interface.c ../net_handler/*.* client/net_handl
 
 ################################ VIRTUAL DEVICES ###############################
 
-devices: general_dev simple_dev unresponsive_dev foreign_dev unstable_dev time_dev sound_dev
+VIRTUAL_DEVICE_FILES := client/virtual_devices/virtual_device_util.* ../dev_handler/message.* ../runtime_util/runtime_util.* ../logger/logger.c
 
-VIRTUAL_DEVICE_FILES := client/virtual_devices/virtual_device_util.o ../dev_handler/message.o ../runtime_util/runtime_util.o ../logger/logger.o
+VIRTUAL_DEVICES := $(shell ls client/virtual_devices/*Device.c | awk -F'.' '{ print $$1 }')
```

Comment:
Uhh idk. I think its something to do with combining shell + Makefile variables. I just used this somewhere else and it worked so I copy pasted it here. It will output everything before the first `.`, so `client/virtual_devices/GeneralTestDevice` which is used to list all devices to make.